sortinghat/core/jobs.py

```diff
@@ -779,7 +780,11 @@ def create_scheduled_task(ctx, job, interval, params):
     if not params:
         params = dict()
 
-    schedule_task(ctx, job_fn, task, **params)
+    try:
+        schedule_task(ctx, job_fn, task, **params)
+    except redis.exceptions.ConnectionError as e:
+        delete_scheduled_task(ctx, task.id)
```

Comment:
I think this is ok but it contradicts the statement on line where we call to `add_scheduled_tasks`. This one is a `db` operation while `delete_scheduled_task` is a api call. For our mental health, I think you should create a `add_schedule_task` in api and replace that call with it.

I think you can remove the `delete` call here and open a new PR where you create that new api function to fix this problem. Do you mind to do that?

Reply:
No problem, I removed the last commit and I'll open a PR with that.